### ISSUE TITLE
[Bug] DateTime Error

### ISSUE THREAD

**Issue by Pascal66** [jeroenterheerdt/HAsmartirrigation#55]
**Describe the bug**
DateTimeError in v0.0.65

**Debug**
This error originated from a custom integration.

Logger: homeassistant
Source: custom_components/smart_irrigation/sensor.py:274
Integration: Smart Irrigation (documentation, issues)
First occurred: 22 avril 2021 à 23:00:01 (1 occurrences)
Last logged: 22 avril 2021 à 23:00:01
Error doing job: Future exception was never retrieved
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/smart_irrigation/sensor.py", line 136, in <lambda>
    lambda event: self._bucket_updated(  # pylint: disable=unnecessary-lambda
  File "/config/custom_components/smart_irrigation/sensor.py", line 300, in _bucket_updated
    self.update_adjusted_run_time_from_event()
  File "/config/custom_components/smart_irrigation/sensor.py", line 274, in update_adjusted_run_time_from_event
    sun_rise = datetime.datetime.strptime(sun_rise, "%Y-%m-%dT%H:%M:%S%z")
  File "/usr/local/lib/python3.8/_strptime.py", line 568, in _strptime_datetime
    tt, fraction, gmtoff_fraction = _strptime(data_string, format)
  File "/usr/local/lib/python3.8/_strptime.py", line 349, in _strptime
    raise ValueError("time data %r does not match format %r" %
ValueError: time data '2021-04-23T04:53:12.253428+00:00' does not match format '%Y-%m-%dT%H:%M:%S%z'
```

**Comment by jeroenterheerdt** [comment]
hi @Pascal66 which version of HA are you on? What type of environment? Also, what date/time format string do you use?

**Comment by Pascal66** [comment]
Hello. Datetime error is only since 2 days, work fine before

```
Version | core-2021.5.0.dev20210422
-- | --
Type d'installation | Home Assistant OS
Développement | true
Supervisor | true
Docker | true
Environnement virtuel | false
Version de Python | 3.8.7
Famille du système d'exploitation | Linux
Version du système d'exploitation | 5.10.28
Architecture du processeur | x86_64
Fuseau horaire | Europe/Paris
```

```
Système d'exploitation hôte | Home Assistant OS 6.0.dev20210419
-- | --
Mise à jour | dev
Version du Supervisor | supervisor-2021.04.dev2301
Version de Docker | 20.10.6
```

**Comment by jeroenterheerdt** [comment]
OK, so what changed about your system? That's the component you should be opening an issue for. I have not updated this component in months.

Get Outlook for Android<https://aka.ms/AAb9ysg>

________________________________

**Comment by Pascal66** [comment]
I'm searching. Will come back when I understand wheres the problem.
Maybe a dev update
Thank you.

**Comment by Heronimonimo** [comment]
I think this bug is still there. I'm having the same issue:
```
2021-05-19 23:00:01 ERROR (MainThread) [homeassistant] Error doing job: Future exception was never retrieved
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/smart_irrigation/sensor.py", line 137, in <lambda>
    lambda event: self._bucket_updated(  # pylint: disable=unnecessary-lambda
  File "/config/custom_components/smart_irrigation/sensor.py", line 301, in _bucket_updated
    self.update_adjusted_run_time_from_event()
  File "/config/custom_components/smart_irrigation/sensor.py", line 275, in update_adjusted_run_time_from_event
    sun_rise = datetime.datetime.strptime(sun_rise, "%Y-%m-%dT%H:%M:%S%z")
  File "/usr/local/lib/python3.8/_strptime.py", line 568, in _strptime_datetime
    tt, fraction, gmtoff_fraction = _strptime(data_string, format)
  File "/usr/local/lib/python3.8/_strptime.py", line 349, in _strptime
    raise ValueError("time data %r does not match format %r" %
ValueError: time data '2021-05-20T03:41:08.470283+00:00' does not match format '%Y-%m-%dT%H:%M:%S%z'
```

Running a supervised HA 2021.5.4 install.

I think it is because new HA versions supply microseconds in the time data. From what I read here https://github.com/home-assistant/core/issues/50525#issuecomment-843583778 changing line 274 in sensor.py to this should fix it:
`sun_rise = datetime.datetime.strptime(sun_rise, "%Y-%m-%dT%H:%M:%S.%f%z")`

**Comment by jeroenterheerdt** [comment]
I am seeing the issue as well now.

**Comment by jeroenterheerdt** [comment]
fixed in v0.0.67
